flight/Modules/Logging/logging.c

```diff
@@ -232,107 +219,92 @@ static void loggingTask(void *parameters)
 			}
 		}
 
-
-		// If currently downloading a log, close the file
-		if (loggingData.Operation == LOGGINGSTATS_OPERATION_LOGGING && read_open) {
-			PIOS_STREAMFS_Close(streamfs_id);
-			read_open = false;
-		}
-
-		if (loggingData.Operation == LOGGINGSTATS_OPERATION_LOGGING && !write_open) {
-			if (PIOS_STREAMFS_OpenWrite(streamfs_id) != 0) {
-				loggingData.Operation = LOGGINGSTATS_OPERATION_ERROR;
-			} else {
-				write_open = true;
+		switch (loggingData.Operation) {
```

Comment:
True, the fact that we can send a wrong state is not very nice. Not sure at the moment how it could be done more elegantly.